src/circuit_proof/prover.rs

```diff
@@ -366,8 +358,22 @@ impl<'a, 'b> ProverCS<'a, 'b> {
 
         let t_x = t_poly.eval(x);
         let t_x_blinding = t_blinding_poly.eval(x);
-        let l_vec = l_poly.eval(x);
-        let r_vec = r_poly.eval(x);
+        let l_vec = l_poly
+            .eval(x)
+            .into_iter()
+            .chain(iter::repeat(Scalar::zero()).take(pad))
+            .collect::<Vec<_>>();
```

Comment:
might be cleaner as `l_poly.eval(x).append(vec![Scalar::zero(); pad]);`

Reply:
thanks, good point